source/js/components/search-form.js

```diff
@@ -0,0 +1,101 @@
+import fetchUtils from "@degjs/fetch-utils";
+
+const searchForm = () => {
+
+
+    const formEl = document.querySelector('.js-search-form');
+    const layoutEl = document.querySelector('.js-general-content')
+
+    const url = formEl.dataset.api;
+
+    function init() {
+        formEl.addEventListener('submit', e => {
+            e.preventDefault();
+            getData();
+        });
+    }
+
+
+    function getData() {
+        const formEls = formEl.querySelectorAll('.js-value');
+        fetchUtils.getJSON(
+            url,
+            {
+                body: JSON.stringify(formEls),
+                method: 'POST',
+                headers: {
+                    'Content-Type': 'application/json'
+                }
+            },
+            {
+                timeout: 10000
+            }
+        )
+        .then((data) => {
+            renderResults(data.results);
+        })
+        .catch(error => {
+            alert(error);
+        });
+    }
+
+    function renderLocations(item) {
+        return `<ul class="doctor-location">
+                    ${item.locations.map(item => {
+                        return `<li class="doctor-location__item">
+                                    <a href="${item.url}" class="doctor-location__link">${item.name}</a>
+                                    <div class="doctor-location__distance">${item.distance.toFixed(1)} Miles</div>
+                                </li>`
+                    }).join('')}
+                </ul>`
+    }
+
+
+    function renderSpecialties(item) {
```

Comment:
Nice job extracting this rendering logic into its own function. Helps readability.